mate-panel/panel-run-dialog.c

```diff
@@ -341,6 +342,18 @@ command_is_executable (const char   *command,
 	if (!result)
 		return FALSE;
 
+	for (int i = 0; i < g_strv_length (argv); i++)
+	{
+		g_autofree gchar *home_path = NULL;
+		g_autofree gchar *tmp_argv = NULL;
+
+		regex = g_regex_new ("^~/", 0, 0, NULL);
+		home_path = g_build_filename (g_get_home_dir (), "/", NULL);
+		tmp_argv = g_regex_replace_literal (regex, argv[i], -1, 0, home_path, 0, NULL);
+		g_free (argv[i]);
+		argv[i] = g_strdup (tmp_argv);
```

Comment:
you could avoid the `g_strdup()` by not marking `tmp_argv` as `g_autofree` (or transfer ownership manually by assigning `NULL` to `tmp_argv` once it has been assigned to `argv[i]`).